scalyr_agent/builtin_monitors/thread_pool.py

```diff
@@ -20,23 +20,38 @@
 from concurrent.futures import ThreadPoolExecutor
 
 from scalyr_agent import scalyr_logging
+import time
 
 global_log = scalyr_logging.getLogger(__name__)
 
 # A MixIn class used for adding a thread poll processing to a BaseServer (i.e. SyslogTCPServer, SyslogUDPServer)
 class ExecutorMixIn:
     def __init__(self, global_config):
-        # Using 4 threads for processing requests, because of GIL and CPU bound tasks higher number would not help process the data faster.
-        reading_threads = 4
+        # Using 8 threads for processing requests, because of GIL and CPU bound tasks higher number would not help process the data faster.
+        # The reason for multiple threads is to avoid deadlock in an unexpected situation before waiting for previous data
+        # on a single request being processed fails and it's not caught do to a bug.
+        processing_threads = 8
```

Comment:
This shouldn't be hardcoded, it makes more sense for it to be some fraction/percentage of the number of available cores

Reply:
The reason for hardcoding it is that it wouldn't help to have more threads since these tasks are CPU bound and because of GIL it wouldn't make it faster. I have set more threads for redundancy only. And it's also configurable.